src/navigation_controller.ts

```diff
@@ -12,43 +12,43 @@
 
 import './gesture_monkey_patch';
 
-import {ASTNode, ShortcutRegistry} from 'blockly/core';
+import * as Blockly from 'blockly/core';
+import {ASTNode, ShortcutRegistry, BlockSvg, WorkspaceSvg} from 'blockly/core';
 import {utils as BlocklyUtils} from 'blockly/core';
 
 import * as Constants from './constants';
 import {Navigation} from './navigation';
 import {Announcer} from './announcer';
+import {LineCursor} from './line_cursor';
+import {BlockCopyData} from 'node_modules/blockly/core/clipboard/block_paster';
 
 /**
  * Class for registering shortcuts for keyboard navigation.
  */
 export class NavigationController {
   /** Data copied by the copy or cut keyboard shortcuts. */
-  copyData = null;
+  copyData: BlockCopyData | null = null;
 
   /** The workspace a copy or cut keyboard shortcut happened in. */
-  copyWorkspace = null;
+  copyWorkspace: WorkspaceSvg | null = null;
+  navigation: Navigation;
+  announcer: Announcer;
 
   /**
    * Constructor used for registering shortcuts.
    * This will register any default shortcuts for keyboard navigation.
    * This is intended to be a singleton.
-   * @param {!Navigation=} optNavigation The class that handles keyboard
-   *     navigation shortcuts. (Ex: inserting a block, focusing the flyout).
    */
-  constructor(optNavigation) {
+  constructor() {
     /**
      * Handles any keyboard navigation shortcuts.
-     * @type {!Navigation}
-     * @public
      */
-    this.navigation = optNavigation || new Navigation();
+    this.navigation = new Navigation();
     this.announcer = new Announcer();
```

Comment:
Can these be constructed in the property declarations (instead of the constructor) actually?

Reply:
Done